### PR TITLE
(Closes #1713) Bug fix for 2D matmul

### LINKED ISSUE #1713: matmul2code transformation produces incorrect code
Embarrassingly, my extension of the matmul2code trans does not produce the correct code for matrix-matrix multiplication.
Currently it gives:

          do j = 1, 3, 1
            do i = 1, 3, 1
              jac(i,j) = 0.0_r_def
              do ii = 1, 3, 1
                jac(i,j) = jac(i,j) + jac_inv(ii,i) * jac_inv_transpose(j,ii)
              enddo
            enddo
          enddo

when it should be:

      jac(i,j) = jac(i,j) + jac_inv(i,ii) * jac_inv_transpose(ii,j)

### PR DESCRIPTION
**Pull request by arporter**

### CODE REVIEW

**Comment by arporter** [comment]
Assuming CI happiness, this is ready for review. It's essentially a two-line fix.

File: src/psyclone/psyir/transformations/intrinsics/matmul2code_trans.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ class Matmul2CodeTrans(Operator2CodeTrans):
             do i=1,P
                 R(i,j) = 0.0
                 do ii=1,N
-                    R(i,j) = R(i,j) + A(ii,i) * B(j,ii)
+                    R(i,j) = R(i,j) + A(i,ii) * B(ii,j)
```
Comment (sergisiso) [inline src/psyclone/psyir/transformations/intrinsics/matmul2code_trans.py:154]: The dimensions of the arrays in the text before the codeblock doesn't look right (* and it also wasn't before the change I think)

**Review by sergisiso — CHANGES_REQUESTED** [review]
@arporter I just have a question about the transformation docstring. The rest looks fine, it is just swapping the symbols and the respective results on the tests. This PR doesn't need any new test coverage of documentations. It can be merged after resolving the question and bringing it to master.

File: src/psyclone/psyir/transformations/intrinsics/matmul2code_trans.py (hunk 1 above)
Comment (arporter) [reply src/psyclone/psyir/transformations/intrinsics/matmul2code_trans.py:154]: Oops, well spotted. I've corrected that now and brought the branch up to master.

**Comment by arporter** [comment]
Ready for another look assuming CI happiness.

**Review by sergisiso — APPROVED** [review]
Now everything looks good. PR approved.

**Pull request merged**